py-polars/tests/unit/functions/as_datatype/test_duration.py

```diff
@@ -143,6 +143,51 @@ def test_add_duration_3786() -> None:
     }
 
 
+def test_datetime_ms_add_duration_us() -> None:
```

Comment:
looks like you've made two tests:
- ms and us
- us and ns

why not parametrise over the time units?

so you'd have something like
```python
@pytest.mark.parametrize(
    ('left_time_unit', 'right_time_unit', 'expected_time_unit'),
    [
        ('ms', 'ms', 'ms'),
        ....
    ]
```

Reply:
Makes sense, will do that

Reply:
@MarcoGorelli

I've parametrised the test with all the possible time units combinations (had to add some helper dictionaries to get the correct multipliers / parameters in each case). 

I've made it to have two cases: a duration added on the right side and vice versa, i.e. datetime + duration & duration + datetime, given that the operations are not "perfectly" commutative. Could be a bit out of scope though (let me know if you think that's too overkill).